core/modules/widgets/widget.js

```diff
@@ -596,11 +596,30 @@ Widget.prototype.findFirstDomNode = function() {
 	return null;
 };
 
+
+
 /*
-Remove any DOM nodes created by this widget or its children
+Depth first destroy
 */
-Widget.prototype.removeChildDomNodes = function() {
-	// If this widget has directly created DOM nodes, delete them and exit. This assumes that any child widgets are contained within the created DOM nodes, which would normally be the case
+Widget.prototype.desendDestroy = function() {
+	this.waitDestroy = true; //for blocking repeat calls to this method from removeChildDomNodes
```

Comment:
No need for this lock, because this method won't get run until it complete its running. JS is run-to-end.

Reply:
some developer may create a destroy() like this:

aWidget.prototype.destroy = function() {
	this.removeChildDomNodes();
        //etc....
};